src/buffer.rs

```diff
@@ -180,6 +193,12 @@ impl<'a, P: Pixel + 'a> Iterator for EnumeratePixels<'a, P> where P::Subpixel: '
     }
 }
 
+impl <'a, P: Pixel + 'a> ExactSizeIterator for EnumeratePixels<'a, P> where P::Subpixel: 'a {
```

Comment:
Can the entire where clause and all bounds on `P` be replaced with `where Self: Iterator`?